src/main/java/com/kodilla/ecommercee/ProductController.java

```diff
@@ -1,4 +1,42 @@
 package com.kodilla.ecommercee;
 
+import com.kodilla.ecommercee.domain.ProductDto;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@RestController
+@RequestMapping("/v1/product")
 public class ProductController {
+
+    @RequestMapping(method = RequestMethod.GET, value = "products")
+    public List<ProductDto> getProducts() {
+        System.out.println("Getting list of products");
+        return new ArrayList<>();
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "product")
+    public ProductDto getProduct(@RequestParam Long productId) {
+        System.out.println("Getting a product by id");
+        return new ProductDto();
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "product", consumes = APPLICATION_JSON_VALUE)
+    public void createProduct(@RequestBody ProductDto productDto) {
+        System.out.println("Creating new product.");
+    }
+
+    @RequestMapping(method = RequestMethod.PUT, value = "update")
+    public ProductDto updateProduct(@RequestBody ProductDto productDto) {
+        System.out.println("Updating the product");
+        return new ProductDto();
+    }
+
+    @RequestMapping(method = RequestMethod.DELETE, value = "product")
```

Comment:
teraz masz url zbudowane:
v1/product/product. Tego value nie trzeba w requestmapping podawać i wtedy zostanie /v1/product.
Przy okazji spróbuj zapisać to przy użyciu @GetMapping itp. to nieco nowsze podejśćie:
https://www.baeldung.com/spring-new-requestmapping-shortcuts

Reply:
Analogicznie dla pozostałych url